xonsh/fs.py

```diff
@@ -0,0 +1,94 @@
+"""
```

Comment:
This is non-lazy in a lot of small ways. Anyway that you can make it more lazy, please?